### DISCUSSION TITLE
Duplicate import inside global and function scopes

### DISCUSSION THREAD

**Discussion by hakancelikdev** [hakancelikdev/unimport#59] [Ideas]
> We can solve this issue by scope analysis.

For example, unused import wrong will be detected in the example below.

```python
import x

def function():
      import x 
      print(x)

print(x)
```

After using unimport

```python
def function():
      import x 
      print(x)

print(x)
```

But, It should be expected codes 

```python
import x 

def function():
      print(x)

print(x)
```

### Discussion

Should we solve this problem? I think this is a very rare case, and I think developer who knows python well does not apply for such use.

**Comment by isik-kaplan** [comment]
Most of the time when an import used inside a function: it is either used for its side effect *when* you call the function, or it is so that it doesn't shadow any other outside variable(which is not really a good reason if you ask me) as far as I've stumbled upon imports like these. 

I honestly wouldn't ever use such a feature since it may potentially break projects inherited -I say inherited as I'd never use imports inside a function- but I'm all in favor of configurable features with smart defaults.

**Comment by hakancelikdev** [comment]
Scope analysis has been added so it works and as expected this discussion can be closed.
